cloud/linode/loadbalancers.go

```diff
@@ -510,18 +511,40 @@ func (l *loadbalancers) getLoadbalancerTags(ctx context.Context, service *v1.Ser
 	return []string{}
 }
 
+func (l *loadbalancers) getLoadbalancerFirewallID(service *v1.Service) (int, error) {
+	fwid, ok := getServiceAnnotation(service, annLinodeCloudFirewallID)
+	if !ok {
+		return -1, nil
+	}
+
+	firewallID, err := strconv.Atoi(fwid)
+	if err != nil {
+		return -1, err
+	}
+	return firewallID, nil
+}
+
 func (l *loadbalancers) createNodeBalancer(ctx context.Context, clusterName string, service *v1.Service, configs []*linodego.NodeBalancerConfigCreateOptions) (lb *linodego.NodeBalancer, err error) {
 	connThrottle := getConnectionThrottle(service)
 
 	label := l.GetLoadBalancerName(ctx, clusterName, service)
 	tags := l.getLoadbalancerTags(ctx, service)
+	firewallID, err := l.getLoadbalancerFirewallID(service)
+	if err != nil {
+		return nil, err
+	}
+
 	createOpts := linodego.NodeBalancerCreateOptions{
 		Label:              &label,
 		Region:             l.zone,
 		ClientConnThrottle: &connThrottle,
 		Configs:            configs,
 		Tags:               tags,
 	}
+
+	if firewallID != -1 {
```

Comment:
I'm not a huge fan of sentinel values. For this to be what I'd expect in idiomatic Go code, it'd have to be either `value, ok, error` as return values of the `getLoadBalancerFirewallID` method. Or the error should be comparable to a "firewall not found" error, so that we know if the return value is usable or not.

Reply:
I'd maybe go as far as inlining the getLoadBalancerFirewallID method here, it's simple enough and it makes error handling more direct.